File: university-service/src/main/java/ua/com/foxminded/courseproject/utils/PageableTransaction.java
package ua.com.foxminded.courseproject.utils;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.data.domain.PageImpl;
import org.springframework.data.domain.Pageable;
import ua.com.foxminded.courseproject.dto.TransactionDto;


import java.util.List;

@Schema(description = "Response-Object Page<StudentDto>).")
public class PageableTransaction extends PageImpl<TransactionDto> {

    @JsonIgnore
    public PageableTransaction(List content, Pageable pageable, long total) {
        super(content, pageable, total);
    }

    @Override
    public String toString() {
        return super.toString();
    }
}

File: university-service/src/main/kotlin/ua/com/foxminded/courseproject/utils/PageableTransaction.kt
package ua.com.foxminded.courseproject.utils

import com.fasterxml.jackson.annotation.JsonIgnore
import io.swagger.v3.oas.annotations.media.Schema
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import ua.com.foxminded.courseproject.dto.TransactionDto

@Schema(description = "Response-Object Page<StudentDto>).")
class PageableTransaction @JsonIgnore constructor(content: List<*>?, pageable: Pageable, total: Long) :
    PageImpl<TransactionDto?>(content as MutableList<TransactionDto?>, pageable, total) {
    override fun toString(): String {
        return super.toString()
    }
}
